packages/web3torrent/src/context/client-initialization-context.ts

```diff
@@ -0,0 +1,37 @@
+import {useState} from 'react';
+import React from 'react';
+import WebTorrentPaidStreamingClient from '../library/web3torrent-lib';
+
+export type InitializationStatus = 'Not Initialized' | 'Initialized' | 'Initializing';
+
+export interface ClientInitializationContextInterface {
+  initialize: () => void;
+  initializationStatus: InitializationStatus;
+}
+
+export const ClientInitializationContext = React.createContext<
+  ClientInitializationContextInterface
+>(undefined);
+
+interface Props {
+  w3tClient: WebTorrentPaidStreamingClient;
+}
+export function useClientInitializationContext({
+  w3tClient
+}: Props): ClientInitializationContextInterface | undefined {
+  const [initializationStatus, setInitializationStatus] = useState<InitializationStatus>(
+    'Not Initialized'
+  );
+
+  const initialize = () => {
+    if (initializationStatus === 'Not Initialized') {
+      setInitializationStatus('Initializing');
+
+      w3tClient.paymentChannelClient.initialize().then(() => {
+        setInitializationStatus('Initialized');
+      });
```

Comment:
This is too intrusive to initialize a member variable of an object for it. 

I think we can do the following:

- Rename `w3tClient.enable()` to `w3tClient.initialize()`
- Call `w3tClient.initialize()` here
- Make sure when we pass in `paymentChannelClient` to build `w3tClient`, `paymentChannelClient` has already been initialized. 
- Add a check to `w3tClient.initialize()` to make sure `paymentChannelClient` is intialized.

What do you think?